hcl2template/types.packer_config.go

```diff
@@ -650,25 +650,10 @@ func (cfg *PackerConfig) GetHCPPackerRegistryBlock() (*HCPPackerRegistryBlock, h
 	var block *HCPPackerRegistryBlock
 	var diags hcl.Diagnostics
 
-	multipleRegistryDiag := func(block *HCPPackerRegistryBlock) *hcl.Diagnostic {
-		return &hcl.Diagnostic{
-			Summary:  "Multiple HCP Packer registry block declaration",
-			Subject:  block.HCL2Ref.DefRange.Ptr(),
-			Severity: hcl.DiagError,
-			Detail: "Multiple " + buildHCPPackerRegistryLabel + " blocks have been found, only one can be defined " +
-				"in HCL2 templates. Starting with Packer 1.12.1, it is recommended to move it to the " +
-				"top-level configuration instead of within a build block.",
-		}
-	}
-	// We start by looking in the build blocks
-	for _, build := range cfg.Builds {
-		if build.HCPPackerRegistry != nil {
-			if block != nil {
-				// error multiple build block
-				diags = diags.Append(multipleRegistryDiag(build.HCPPackerRegistry))
-				continue
-			}
-			block = build.HCPPackerRegistry
+	// build block leveled hcp registry is only tolerated when there is only one build block
+	if len(cfg.Builds) <= 1 {
```

Comment:
Likely not an issue, but if we don't have a build block defined here, what happens? I'm not sure we even reach this code, but I wonder if the check here shouldn't be exactly `len(cfg.Builds) == 1` instead of a `<=`, otherwise we'll crash on the following statement since `cfg.Builds[0]` will be out-of-bounds